cmd/minishift/cmd/image/config_add.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"github.com/spf13/cobra"
+
+	"fmt"
+	"github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/minishift/minishift/pkg/util/strings"
+)
+
+var (
+	addConfiguredImageCmd = &cobra.Command{
+		Use:   "add [image ...]",
+		Short: "Adds the specified images to the list of configured images for import and export.",
+		Long:  "Adds the specified images to the list of configured images for import and export.",
+		Run:   addConfiguredImage,
+	}
+)
+
+func addConfiguredImage(cmd *cobra.Command, args []string) {
+	if len(args) == 0 {
+		atexit.ExitWithMessage(1, noImageSpecified)
+	}
+
+	configMap, err := config.ReadConfig()
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Cannot read the Minishift configuration: %s", err.Error()))
+	}
+
+	var cacheImages []string
+	if configMap[config.CacheImages.Name] == nil {
+		cacheImages = []string{}
+	} else {
+		cacheImages = toStringSlice(configMap[config.CacheImages.Name].([]interface{}))
+	}
+
+	for _, image := range args {
```

Comment:
This is general initialization code:
```
+	var cacheImages []string
+	if configMap[config.CacheImages.Name] == nil {
+		cacheImages = []string{}
+	} else {
+		cacheImages = toStringSlice(configMap[config.CacheImages.Name].([]interface{}))
+	}
```
which can remain part of the append/add function, but this is considered utility:
```
+	for _, image := range args {
+		if !strings.Contains(cacheImages, image) {
+			cacheImages = append(cacheImages, image)
+		}
+	}
```